web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.tsx

```diff
@@ -73,235 +89,291 @@ export type Props = {
-        askDenom: currency?.askDenom,
-        cryptoSellOrders,
-        retiring,
-      }),
-    [cryptoSellOrders, currency?.askDenom, retiring],
-  );
+    const filteredCryptoSellOrders = useMemo(
+      () =>
+        getFilteredCryptoSellOrders({
+          askDenom: currency?.askDenom,
+          cryptoSellOrders,
+          retiring,
+        }),
+      [cryptoSellOrders, currency?.askDenom, retiring],
+    );
 
-  const orderedSellOrders = useMemo(
-    () =>
-      card
-        ? cardSellOrders.sort((a, b) => a.usdPrice - b.usdPrice)
-        : filteredCryptoSellOrders?.sort(
-            (a, b) => Number(a.askAmount) - Number(b.askAmount),
-          ) || [],
+    const orderedSellOrders = useMemo(
+      () =>
+        card
+          ? cardSellOrders.sort((a, b) => a.usdPrice - b.usdPrice)
+          : filteredCryptoSellOrders?.sort(
+              (a, b) => Number(a.askAmount) - Number(b.askAmount),
+            ) || [],
 
-    [card, cardSellOrders, filteredCryptoSellOrders],
-  );
+      [card, cardSellOrders, filteredCryptoSellOrders],
+    );
 
-  const handleCryptoPurchaseOptions = useCallback(() => {
-    setRetiring(prev => !prev);
-  }, [setRetiring]);
+    const handleCryptoPurchaseOptions = useCallback(() => {
+      setRetiring(prev => !prev);
+    }, [setRetiring]);
 
-  const handlePaymentOptions = useCallback(
-    (option: string) => {
-      setPaymentOption(option as PaymentOptionsType);
-      form.setValue(CREDIT_VINTAGE_OPTIONS, []);
-      form.setValue(
-        CURRENCY,
-        option === PAYMENT_OPTIONS.CARD ? cardCurrency : defaultCryptoCurrency,
-      );
-    },
-    [setPaymentOption, form, cardCurrency, defaultCryptoCurrency],
-  );
+    const handlePaymentOptions = useCallback(
+      (option: string) => {
+        startTransition(() => {
+          setPaymentOption(option as PaymentOptionsType);
+        });
+        const currency =
+          option === PAYMENT_OPTIONS.CARD
+            ? cardCurrency
+            : defaultCryptoCurrency;
+        form.reset({
+          ...form.getValues(),
+          [CREDIT_VINTAGE_OPTIONS]: [],
+          [CURRENCY]: currency,
+        });
+        updateMultiStepCurrencyAndPaymentOption(
+          handleSave,
+          data,
+          currency,
+          activeStep,
+          option,
+        );
+      },
+      [
+        form,
+        cardCurrency,
+        defaultCryptoCurrency,
+        handleSave,
+        data,
+        activeStep,
+        setPaymentOption,
+      ],
+    );
 
-  useEffect(() => {
-    // If there are some sell orders available for fiat purchase, default to 'card' option
-    if (cardSellOrders.length > 0 && !initialPaymentOption) {
-      setPaymentOption(PAYMENT_OPTIONS.CARD);
-      form.setValue(CREDIT_VINTAGE_OPTIONS, []);
-      form.setValue(CURRENCY, cardCurrency);
-    }
-  }, [cardSellOrders.length, initialPaymentOption]); // just run this once
+    useEffect(() => {
+      if (spendingCap === null) {
```

Comment:
I don't believe this is necessarily needed, `setSpendingCap` is already set in `CreditsAmount`
just checking `spendingCap === null` in `createChooseCreditsFormSchema` seems to fix the issue above without this